Using payment intents to make payments

mutthias opened the pull request:
If needed I can of course walk through in a call what I did/how what I did works. 

(Assuming this is Austin reviewing) This works from /checkout. I have a super basic credit card info form to make payments since the checkout form already has stuff like address. From last time we created a checkout form on the backend with:

`const session = await auth_stripe.checkout_session.create({})`

But for this, I swapped it (as well as the params) to:

`const session = await auth_stripe.paymentIntents.create`

Everything seems to be plug and play. I hardcoded billing info name, email, as well as the amounts for the payment intent, but when it all goes through I console.log everything from the payment intent. The payment intent doesn't directly give me the payment method, but it gives me a payment method ID, which I used to fetch payment methods to get stuff like the last 4 digits of someone's credit card (this is all console.logged).

mutthias committed:
Payment intents instead of product page

mutthias committed:
Return payment method

Austin2Shih reviewed (approved):
This looks good! There are a few things we can do to clean up the code but I think it'll be best if you start working on integrating your stuff with the GraphQL API to be able to update order status or like createOrders when we get pinged. Both the CreateOrder and UpdateOrder stuff should already be made.

You can start with unhardcoding this stuff first though.

mutthias committed:
Create payment amounts from shopping cart

mutthias committed:
Removed some unneeded console.logs

mutthias committed:
Unhardcoded billing details

A few things:
- Moved PaymentForm to ShippingInfo.tsx to be able to pass billing info as props
- The billing info form is wrapped as a div instead of a form to prevent nested forms, and also because there's no button connected to it
- A few components have been changed to accept a handful of props for billing info

In summary, the billing info is not hardcoded and uses whatever information is inputted in our form 👍

mutthias committed:
GraphQL mutation and queries for orders

Just wrote out a mutation for posting billing info to our own database rathe than having to hop on Stripe to manage orders. This is done ONLY after the order successfully goes through Stripe.

Also I made two queries for grabbing all the orders or just one order from our database.

Only thing is that everything isn't tested because I couldn't see if my branch had a way for me to run a server (if that's setup already)

(Austin or Brandon please save me tomorrow in meeting if you see this before)
